### PR TITLE
BLD: [WIP] Adding misc to meson.build

### PR DESCRIPTION
**Pull request by Smit-create**
#### Additional information

Adding `misc` to `meson.build`

cc @rgommers

### CODE REVIEW

File: scipy/misc/meson.build [hunk 1]
```diff
@@ -0,0 +1,18 @@
+python_sources = [
+  '__init__.py',
+  'common.py',
+  'doccer.py',
+]
+
+py3.install_sources(
+    python_sources,
+    pure: false,
+    subdir: 'scipy/misc'
```
Comment (rgommers) [inline scipy/misc/meson.build:10]: minor: indentation level in `meson.build` files should be 2 spaces. 

we can fix this later in a style cleanup, that will be needed before we open a PR to SciPy for adding the full build system.

**Review by rgommers — APPROVED** [review]
Thanks @Smit-create, works like a charm!

**Comment by rgommers** [comment]
One other small thing: SciPy (and NumPy) have a commit message convention with prefixes, as documented in https://docs.scipy.org/doc/numpy/dev/development_workflow.html#writing-the-commit-message. Would be nice to use that in future commits.

File: scipy/misc/meson.build (hunk 1 above)
Comment (Smit-create) [reply scipy/misc/meson.build:10]: Sure!, I will keep that in my mind next time

**Pull request merged**